src/intermediate_config.rs

```diff
@@ -8,19 +8,24 @@ use self::yaml_rust::YamlLoader;
 
 #[derive(Debug, PartialEq, Eq)]
 pub struct IntermediateConfig {
-    pub remote_machine: Option<IntermediateRemoteMachine>,
-    pub compression: Option<IntermediateCompression>,
+    pub remote: Option<IntermediateRemote>,
+    pub push: Option<IntermediatePush>,
+    pub pull: Option<IntermediatePull>,
```

Comment:
Really weird, but I like that both `push` and `pull` have the same horizontal space...